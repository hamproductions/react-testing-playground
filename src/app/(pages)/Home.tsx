'use client';

import { Suspense } from 'react';
import { Calculator } from '../components/calculator/Calculator';
import { useCalculator } from '../components/calculator/useCalculator';
import { Weather } from '../components/weather/Weather';
import { Text } from 'components/ui/text';
import { Heading } from 'components/ui/heading';
import { Center, Stack, HStack } from 'styled-system/jsx';

export function Home() {
  const MAX_LENGTH = 10;
  const { equation, onButtonPressed } = useCalculator({ maxLength: MAX_LENGTH });
  return (
    <Center flex={1} w="full" minH="screen">
      <Stack alignItems="center" w="full">
        <Heading fontSize="2xl">計算機 v2.0</Heading>
        {equation.length >= MAX_LENGTH && <Text>ごめんなさい式は{MAX_LENGTH}までです</Text>}
        <HStack justifyContent="space-around" alignItems="stretch" w="full">
          <Calculator equation={equation} onButtonPressed={onButtonPressed} />
        </HStack>
        <Suspense fallback={<Text>読み込み中…</Text>}>
          <Weather />
        </Suspense>
      </Stack>
    </Center>
  );
}
