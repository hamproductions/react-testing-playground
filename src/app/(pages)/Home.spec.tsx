import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import userEvent from '@testing-library/user-event';
import { Home } from './Home';

import '@testing-library/jest-dom/vitest';

describe('Home Page', () => {
  describe('Calculator', () => {
    it('Match Snapshot', () => {
      const { container } = render(<Home />);
      expect(container).toMatchSnapshot();
    });

    it('plus', () => {
      const { getByText } = render(<Home />);

      expect(getByText('()')).toBeDisabled();
    });

    it('plus', async () => {
      const { getByText } = render(<Home />);

      const user = userEvent.setup();
      const ichibotan = getByText('1');
      await user.click(ichibotan);
      await user.click(ichibotan);
      await user.click(getByText('+'));
      await user.click(ichibotan);
      await user.click(getByText('='));

      expect(getByText('12')).toBeVisible();
    });

    it('minus', async () => {
      const { getByText } = render(<Home />);

      const user = userEvent.setup();
      const ichibotan = getByText('1');
      await user.click(ichibotan);
      await user.click(ichibotan);
      await user.click(getByText('-'));
      await user.click(ichibotan);
      await user.click(getByText('='));

      expect(getByText('10')).toBeVisible();
    });

    it('minus', async () => {
      const { getByText } = render(<Home />);

      const user = userEvent.setup();
      const ichibotan = getByText('1');
      await user.click(ichibotan);
      await user.click(ichibotan);
      await user.click(getByText('×'));
      await user.click(ichibotan);
      await user.click(ichibotan);
      await user.click(getByText('='));

      expect(getByText('121')).toBeVisible();
    });
  });
  describe('Weather Report', () => {
    it('fetch', async () => {
      const { findByText } = render(<Home />);

      expect(await findByText('99°C')).toBeVisible();
    });
  });
});
