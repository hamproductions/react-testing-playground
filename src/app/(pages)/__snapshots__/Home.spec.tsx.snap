// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page > Calculator > Match Snapshot 1`] = `
<div>
  <div
    class="d_flex ai_center jc_center flex_1 w_full min-h_screen"
  >
    <div
      class="d_flex flex-d_column ai_center gap_10px w_full"
    >
      <h2
        class="text text--variant_heading fs_2xl"
      >
        計算機 v2.0
      </h2>
      <div
        class="d_flex ai_stretch jc_space-around gap_10px flex-d_row w_full"
      >
        <div
          class="bd_1px_solid bd-c_border.default bdr_l1 p_4"
        >
          <div
            class="d_flex flex-d_column gap_4"
          >
            <div
              class="bdr_l2 p_1 bg-c_bg.muted"
            >
              <p
                class="text ta_end fs_5xl"
              >
                0
              </p>
            </div>
            <div
              class="d_grid grid-tc_repeat(4,_1fr) gap_10px"
            >
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                AC
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
                disabled=""
              >
                ()
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
                disabled=""
              >
                %
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                ÷
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                7
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                8
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                9
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                +
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                4
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                5
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                6
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                -
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                1
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                2
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                3
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                ×
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                0
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
                disabled=""
              >
                .
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                ←
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
      <p
        class="text"
      >
        読み込み中…
      </p>
    </div>
  </div>
</div>
`;

exports[`Home Page > Calculator > plus 1`] = `
<div>
  <div
    class="d_flex ai_center jc_center flex_1 w_full min-h_screen"
  >
    <div
      class="d_flex flex-d_column ai_center gap_10px w_full"
    >
      <h2
        class="text text--variant_heading fs_2xl"
      >
        計算機 v2.0
      </h2>
      <div
        class="d_flex ai_stretch jc_space-around gap_10px flex-d_row w_full"
      >
        <div
          class="bd_1px_solid bd-c_border.default bdr_l1 p_4"
        >
          <div
            class="d_flex flex-d_column gap_4"
          >
            <div
              class="bdr_l2 p_1 bg-c_bg.muted"
            >
              <p
                class="text ta_end fs_5xl"
              >
                0
              </p>
            </div>
            <div
              class="d_grid grid-tc_repeat(4,_1fr) gap_10px"
            >
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                AC
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
                disabled=""
              >
                ()
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
                disabled=""
              >
                %
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                ÷
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                7
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                8
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                9
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                +
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                4
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                5
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                6
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                -
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                1
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                2
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                3
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                ×
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                0
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
                disabled=""
              >
                .
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                ←
              </button>
              <button
                class="button button--variant_solid button--size_xl bdr_xl fs_xl"
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
      <p
        class="text"
      >
        読み込み中…
      </p>
    </div>
  </div>
</div>
`;
